e2e/cypress/integration/enterprise/guest_accounts/guest_invitation_ui_spec.js

```diff
@@ -0,0 +1,121 @@
+            cy.wrap($el).find('.arrow').click();
+        });
+
+        // * Verify the header has changed in the modal
+        cy.getByTestId('invitationModal').within(($el) => {
+            cy.wrap($el).find('h1').should('have.text', 'Invite Guests');
+        });
+
+        // *Verify Invite Guests button is disabled by default
+        cy.get('#inviteGuestButton').scrollIntoView().should('be.visible').and('be.disabled');
+
+        // *Verify Invite People field
+        const email = `temp-${getRandomInt(9999).toString()}@mattermost.com`;
+        cy.getByTestId('addPeople').should('be.visible').within(($el) => {
+            cy.wrap($el).find('h2 > span').should('have.text', 'Invite People');
+            cy.wrap($el).find('.help-text > span').should('have.text', 'Search and add guests or email invite new users.');
+        });
+        cy.getByTestId('emailPlaceholder').should('be.visible').within(($el) => {
+            // * Verify the input placeholder text
+            cy.wrap($el).get('.users-emails-input__placeholder').should('have.text', 'Add guests or email addresses');
+
+            // # Type the email of the new user
+            cy.wrap($el).get('input').type(email, {force: true});
+            cy.wrap($el).get('.users-emails-input__menu').
+                children().should('have.length', 1).
+                eq(0).should('contain', `Invite ${email} as a guest`).click();
+        });
+
+        // *Verify Search and Add Channels
+        cy.getByTestId('channelPlaceholder').should('be.visible').within(($el) => {
+            cy.wrap($el).find('h2 > span').should('have.text', 'Search and Add Channels');
+            cy.wrap($el).find('.help-text > span').should('have.text', 'Specify the channels the guests have access to.');
+        });
+        cy.getByTestId('channelPlaceholder').should('be.visible').within(($el) => {
+            // * Verify the input placeholder text
+            cy.wrap($el).get('.channels-input__placeholder').should('have.text', 'Search and add channels');
+
+            // # Type the channel name
+            cy.wrap($el).get('input').type('town sq', {force: true});
+            cy.wrap($el).get('.channels-input__menu').
+                children().should('have.length', 1).
+                eq(0).should('contain', 'Town Square').click();
+        });
+
+        // *Verify Set Custom Message before clicking on the link
+        cy.getByTestId('customMessage').should('be.visible').within(($el) => {
+            cy.wrap($el).get('#customMessageHeader').should('not.exist');
+            cy.wrap($el).get('textarea').should('not.exist');
+            cy.wrap($el).get('.help-text').should('have.text', 'Create a custom message to make your invite more personal.');
+
+            // #Verify link text and click on it
+            cy.wrap($el).get('a').should('have.text', 'Set a custom message').click();
```

Comment:
nit: Can we have href check as well?

Reply:
Any reason we need to have this? We are anyways trying to click on it and verify the behavior.

Reply:
0/5, no particular reason.  We're using anchor element in many ways: opening a link, direct to internal target, etc. and for some of those, we're checking href.

Reply:
Thanks. It makes sense to validate href for external links. But for onclick actions, it's better to validate the behavior when a link is clicked.